lib/internal/streams/from.js

```diff
@@ -73,13 +74,9 @@ function from(Readable, iterable, opts) {
 
   async function next() {
     try {
-      needToClose = false;
       const { value, done } = await iterator.next();
-      needToClose = !done;
       if (done) {
         readable.push(null);
-      } else if (readable.destroyed) {
-        await close();
```

Comment:
This check is no longer necessary without needToClose  since `_destroy` is called and if the stream is destroyed the iterator is closed.